### ISSUE TITLE
Incorrect Dependencies might be used during build stage

### ISSUE THREAD

**Issue by azlam-abdulsalam** [dxatscale/sfpowerscripts#496]
**Describe the bug**
When packages are being built in parallel from different branches at the same time and semantic version is not correctly followed. This would result in packages being built using last known validated packages from a different branch. 

**To Reproduce**
1. Run an orchestrator:build on develop
2. Run an orchestrator:build on release

Ensure the packages to be built are same and also its dependencies need to be built. Also ensure semantic version is not followed, where build and release is using the same major.minor.patch version and entirely dependent on the 'next' segment

**Expected behavior**
Only should utilize dependencies from the particular branch or promoted ones if the dependent packages are built from another repo.

**Comment by azlam-abdulsalam** [comment]
To fix this issue, sfpowerscripts will enforce the branch flag in build command. This will break any existing pipelines and new build of all package (from all the open branches) is also required as the there is internal artifact changes

**Comment by azlam-abdulsalam** [comment]
Moving to milestone 23, to get sfpowerkit to accomodate the change

**Comment by azlam-abdulsalam** [comment]
@jhawk-4  Thanks for pointing the fact that .NEXT resets to zero, when this goes live. We might meed to get everyone to do a patch update on all packages

**Comment by azlam-abdulsalam** [comment]
This is the workflow for resolving package dependencies

![image](https://user-images.githubusercontent.com/43767972/166344669-66426692-2a9f-45ce-b5ad-238f32e121bd.png)

**Comment by vuha-acn** [comment]
@azlam-abdulsalam Follow-up questions:
1. Does this account for both unlocked and source packages (eg. DevHub would be used to track the source package last validated version stored in the Sfpowerscripts artifacts Custom Settings Object?
2. If the dependent package is not part of the current repo, what happens if it's not available in the same DevHub and another DevHub?

Can we provide a more concrete example using Package A Version 1.0.0, Package B Version 1.1.0 in a diagram to provide more concrete examples? I'm still a little unclear on the versions retrieved from the DevHub for dependencies.

**Comment by azlam-abdulsalam** [comment]
1. This is applicable only for unlocked packages.
2. It doesn't matter, if its not part of the repo, we treat it like any other external package ( we use the provided 04t Id)

This is basically enhancing how we are resolving .LATEST to the correct build number
